packages/@ember/-internals/glimmer/tests/integration/helpers/hash-test.js

```diff
@@ -217,92 +217,5 @@ moduleFor(
 
       this.assertText('Godfrey Chan');
     }
-
-    ['@test works with computeds on non-defined properties']() {
-      let instance;
-
-      let FooBarComponent = Component.extend({
-        init() {
-          this._super(...arguments);
-
-          expectDeprecation(() => {
-            set(this.hash, 'lastName', 'Hietala');
-          }, /You set the '.*' property on a {{hash}} object/);
-
-          instance = this;
-        },
-
-        fullName: computed('hash.firstName', 'hash.lastName', function () {
-          return `${this.hash.firstName} ${this.hash.lastName}`;
-        }),
-      });
-
-      this.registerComponent('foo-bar', {
-        ComponentClass: FooBarComponent,
-        template: `{{this.fullName}}`,
-      });
-
-      this.render(`{{foo-bar hash=(hash firstName=this.firstName)}}`, {
-        firstName: 'Chad',
-        lastName: 'Hietala',
-      });
-
-      this.assertText('Chad Hietala');
-
-      runTask(() => this.rerender());
-
-      this.assertText('Chad Hietala');
-
-      runTask(() => {
-        set(this.context, 'firstName', 'Godfrey');
-      });
-
-      // needs to be separate task because of the way classic components update args
-      runTask(() => {
-        expectDeprecation(() => {
-          set(instance.hash, 'lastName', 'Chan');
-        }, /You set the '.*' property on a {{hash}} object/);
-      });
-
-      this.assertText('Godfrey Chan');
-    }
-
-    ['@test works when properties are set dynamically']() {
-      let fooBarInstance;
-      let FooBarComponent = Component.extend({
-        init() {
-          this._super();
-          fooBarInstance = this;
-        },
-      });
-
-      this.registerComponent('foo-bar', {
-        ComponentClass: FooBarComponent,
-        template: `{{this.hash.firstName}} {{this.hash.lastName}}`,
-      });
-
-      this.render(`{{foo-bar hash=(hash firstName=this.firstName)}}`, {
-        firstName: 'Chad',
-      });
-
-      this.assertText('Chad ');
-
-      runTask(() => {
-        expectDeprecation(() => {
-          set(fooBarInstance.hash, 'lastName', 'Hietala');
-        }, /You set the '.*' property on a {{hash}} object/);
-      });
-
-      this.assertText('Chad Hietala');
-
-      runTask(() => {
-        expectDeprecation(() => {
-          set(fooBarInstance.hash, 'firstName', 'Godfrey');
-          set(fooBarInstance.hash, 'lastName', 'Chan');
-        }, /You set the '.*' property on a {{hash}} object/);
-      });
-
-      this.assertText('Godfrey Chan');
-    }
```

Comment:
This went from deprecation to nothing (instead of an assertion). I don't think it's illegal per-se, I think we are technically allowed to do it, but that's when we tend to get ourselves into trouble – see the `{{#with}}` thing recently. Next time at least consider whether we can keep an assertion around for at least 1-2 LTS.